core/src/either.rs

```diff
@@ -166,13 +172,13 @@ where
             EitherOutput::First(inner) => {
                 match substream {
                     EitherOutbound::A(substream) => inner.destroy_outbound(substream),
-                    _ => panic!("Wrong API usage")
+                    _ => panic!("Wrong API usage, a substream is never destroyed")
```

Comment:
So I think what this case is about is when we get an `EitherOutput::First` and an `EitherOutbound:B`, which should (must) never happen, so the message here is not correct. Maybe "`panic!("Wrong API usage: impossible combination of Output/Outbounds")` (or some such).
@twittner thoughts?

Reply:
Makes sense, just read the destroy_outbound for the inner `StreamMuxer` as well, was also wrong that a substream is never destroyed.